cli/pcluster/config/validators.py

```diff
@@ -0,0 +1,629 @@
+    errors = []
+    warnings = []
+    try:
+        test = boto3.client("ec2").describe_snapshots(SnapshotIds=[param_value]).get("Snapshots")[0]
+        if test.get("State") != "completed":
+            warnings.append("Snapshot {0} is in state '{1}' not 'completed'".format(param_value, test.get("State")))
+    except ClientError as e:
+        errors.append(e.response.get("Error").get("Message"))
+
+    return errors, warnings
+
+
+def ec2_volume_validator(param_key, param_value, pcluster_config):
+    errors = []
+    warnings = []
+    try:
+        test = boto3.client("ec2").describe_volumes(VolumeIds=[param_value]).get("Volumes")[0]
+        if test.get("State") != "available":
+            warnings.append("Volume {0} is in state '{1}' not 'available'".format(param_value, test.get("State")))
+    except ClientError as e:
+        if e.response.get("Error").get("Message").endswith("parameter volumes is invalid. Expected: 'vol-...'."):
+            errors.append("Volume {0} does not exist".format(param_value))
+        else:
+            errors.append(e.response.get("Error").get("Message"))
+
+    return errors, warnings
+
+
+def efs_validator(section_key, section_label, pcluster_config):
+    errors = []
+    warnings = []
+
+    section = pcluster_config.get_section(section_key, section_label)
+    throughput_mode = section.get_param_value("throughput_mode")
+    provisioned_throughput = section.get_param_value("provisioned_throughput")
+
+    if throughput_mode != "provisioned" and provisioned_throughput:
+        errors.append("When specifying 'provisioned_throughput', the 'throughput_mode' must be set to 'provisioned'")
+
+    if throughput_mode == "provisioned" and not provisioned_throughput:
+        errors.append(
+            "When specifying 'throughput_mode' to 'provisioned', the 'provisioned_throughput' option must be specified"
+        )
+
+    return errors, warnings
+
+
+def raid_volume_iops_validator(param_key, param_value, pcluster_config):
+    errors = []
+    warnings = []
+
+    raid_iops = float(param_value)
+    raid_vol_size = float(pcluster_config.get_section("raid").get_param_value("volume_size"))
+    if raid_iops > raid_vol_size * 50:
+        errors.append("IOPS to volume size ratio of {0} is too high; maximum is 50.".format(raid_iops / raid_vol_size))
+
+    return errors, warnings
+
+
+def scheduler_validator(param_key, param_value, pcluster_config):
+    errors = []
+    warnings = []
+
+    if param_value == "awsbatch":
+        if pcluster_config.region in [
+            "ap-northeast-3",
+            "eu-north-1",
+            "cn-north-1",
+            "cn-northwest-1",
+            "us-gov-east-1",
+            "us-gov-west-1",
+        ]:
+            errors.append("'awsbatch' scheduler is not supported in the '{0}' region".format(pcluster_config.region))
+
+    return errors, warnings
+
+
+def cluster_validator(section_key, section_label, pcluster_config):
```

Comment:
can we move these validations to the specific parameters: min_vcpus, desired_vcpus, etc?

Reply:
We could but it would be a repetitive test because:
- for min we should compare with desired and max
- for desired we should check max and min
- for max ... etc

In this way we are doing the comparison in a single place.